### PR TITLE
Send binary LLSD to LEAP plugins; extend LLSD compatibility, redux

### PR DESCRIPTION
**Pull request by AndrewMeadows**
This PR ports work from #29 (which was put in the contribute branch) for the DRTVWR-558 branch (puppetry). It is not clear if the work piling up in contribute will ever be merged, and enhancing the LEAP channel is of particular interest to the puppetry project, so this port is being submitted in case we decide to use it.

Note: with the puppetry project experiencing a long development cycle on its own branch... finally merging it into the main branch will be a messy operation and the work in this PR will contribute to that mess significantly. However, perhaps that will be the best path for the final delivery of this work.

### CODE REVIEW

**Review by nat-goodspeed — CHANGES_REQUESTED** [review]
I'm happy with the set of commits. I do spot a few things that I think should be improved (as commented inline). Are you willing to make those changes? Let me know if you want me to do it instead.

**Comment by nat-goodspeed** [comment]
Or would it be better for me to make the changes I requested as a PR against `contribute`, and you can cherry-pick those into this branch?

File: indra/llcommon/llkeybind.cpp [hunk 1]
```diff
@@ -213,19 +214,23 @@ bool LLKeyBind::isEmpty() const
...
+    return std::find_if_not(mData.rbegin(), mData.rend(),
+                            [](const auto& kdata){ return kdata.empty(); }).base();
+}
 
+LLSD LLKeyBind::asLLSD() const
+{
     LLSD data;
-    for (S32 i = 0; i <= last; ++i)
+    auto end{ endNonEmpty() };
+    for (auto it = mData.begin(); it < end; ++it)
```
[the remark this answers is not in the document]
Comment (nat-goodspeed) [reply indra/llcommon/llkeybind.cpp:229]: Interesting - a Fawrsk on the `contribute` branch replaced this with a range `for`, eliminating the implicit trimming of empty tail entries. I won't change it back, meaning that if you want to change it on your branch, you'll have to tweak it yourself.

File: indra/llcommon/llsys_objc.mm [hunk 2]
```diff
@@ -29,17 +29,17 @@
...
 {
     if (NSAppKitVersionNumber > NSAppKitVersionNumber10_8)
     {
         NSOperatingSystemVersion osVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
-        major = osVersion.majorVersion;
-        minor = osVersion.minorVersion;
-        patch = osVersion.patchVersion;
+        major = int(osVersion.majorVersion);
+        minor = int(osVersion.minorVersion);
+        patch = int(osVersion.patchVersion);
```
[the remark this answers is not in the document]
Comment (nat-goodspeed) [reply indra/llcommon/llsys_objc.mm:42]: Already fixed on `contribute` branch

**Comment by brad-linden** [comment]
recheck

**Comment by nat-goodspeed** [comment]
@AndrewMeadows, please merge in the changesets from https://github.com/secondlife/viewer/pull/99. Thanks!

**Review by nat-goodspeed — APPROVED** [review]

**Pull request merged**